CMakeLists.txt

```diff
@@ -1,4 +1,15 @@
-cmake_minimum_required(VERSION 3.1.3)
+cmake_minimum_required(VERSION 3.13)
+
+# These linker flags must be set before the project() command below, because
+# linker flags set later are not passed to CMake's test compiles at configure
+# time. On Windows, the linker must be overridden with -DCMAKE_LINKER= because
+# it is invoked directly by CMake rather than by the compiler.
+option(BYN_ENABLE_LTO "Build with LTO" Off)
+if(BYN_ENABLE_LTO AND NOT MSVC)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=lld")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fuse-ld=lld")
```

Comment:
Should we also check that the compiler is clang (and not gcc)?

Reply:
I think `-fuse-ld=lld` works with gcc as well: https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html

Reply:
But the object files produced by GCC LTO are not compatible with LLVM LTO and vice versa.  So if you compile with gcc and link with lld there is no way your LTO object files will be useable.   i.e. if you use LTO you compiler and linker have to match.

Reply:
Oh, good point @sbc100 , I didn't know that but it makes complete sense...

Reply:
Well you can actually use LLVM LTO with GNU ld or gold (if the LLVM LTO plugin that matches clang is installed). But not vice versa (I don't think?).